app/controllers/shared/api.php

```diff
@@ -114,6 +115,27 @@
 
     $deletes->setProject($project);
     $database->setProject($project);
+
+    $useCache = $route->getLabel('useCache', false);
+    if ($useCache) {
+            $key = md5($request->getURI() . $request->getServer('query_string'));
+            $cache = new Cache(new Filesystem(APP_STORAGE_CACHE . DIRECTORY_SEPARATOR . 'app-' . $project->getId()));
+            $data = $cache->load($key, 60 * 60 * 24 * 30 * 1);
+        if (!empty($data)) {
+            $cacheLog = $dbForProject->getDocument('cache', $key);
+            if ($cacheLog->isEmpty()) {
+                Authorization::skip(fn () => $dbForProject->createDocument('cache', new Document([
+                        '$id' => $key,
+                        'accessedAt' => time(),
+                    ])));
+            } else {
+                $cacheLog->setAttribute('accessedAt', time());
+                Authorization::skip(fn () => $dbForProject->updateDocument('cache', $cacheLog->getId(), $cacheLog));
+            }
+            $data = json_decode($data, true);
+            $response->file(base64_decode($data['payload']), $data['content-type'], $data['date'], 'hit');
```

Comment:
Here we can just set the "state" header with `$response->header()`.

Reply:
fixed